# This patch fixes a Solaris specific issue related to incorrect
# reporting of network interface speed inside a zone. The patch has
# been submitted to the community but has not been accepted yet.

# The details can be found in the following location
# https://sourceforge.net/p/net-snmp/patches/1277/
diff -wpruN '--exclude=*.orig' net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c net-snmp-5.8/agent/mibgroup/kernel_sunos5.c
--- net-snmp-5.8~/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:41.879550292 +0000
+++ net-snmp-5.8/agent/mibgroup/kernel_sunos5.c	2018-07-24 09:51:42.286675297 +0000
@@ -1949,8 +1949,10 @@ set_if_info(mib2_ifEntry_t *ifp, unsigne
 	havespeed = B_TRUE;
     } else if (getKstatInt("link", name, "ifspeed", &ifp->ifSpeed) == 0) {
 	havespeed = B_TRUE;
+    } else if ((getKstatInt("link", name, "ifspeed", &ifp->ifSpeed) == 0) ||
+               (getKstatInt("link", name, "ifSpeed", &ifp->ifSpeed) == 0)) {
+        havespeed = B_TRUE;
     }
-
     /* make ifOperStatus depend on link status if available */
     if (ifp->ifAdminStatus == 1) {
         int i_tmp;
